src/parquet/column/scanner.h

```diff
@@ -0,0 +1,208 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PARQUET_COLUMN_SCANNER_H
+#define PARQUET_COLUMN_SCANNER_H
+
+#include <memory>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "parquet/column/reader.h"
+#include "parquet/thrift/parquet_types.h"
+
+namespace parquet_cpp {
+
+class ColumnReader;
+
+template <int TYPE>
+class TypedColumnReader;
+
+#define DEFAULT_SCANNER_BATCH_SIZE 128
+
+class Scanner {
+ public:
+  explicit Scanner(std::shared_ptr<ColumnReader> reader) :
+      reader_(reader),
+      batch_size_(DEFAULT_SCANNER_BATCH_SIZE),
```

Comment:
Instead of pre-defining the default batch size, it is better to make it a ctor argument with a default value 128.